vb/sys_user_manager

kreanimator opened the pull request:
Created sys_user_manager and SAM templates to it

kreanimator committed:
Created sys_user_manager and SAM templates to it

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,63 @@
+    def __call__(self, event):
+
+        logger.info("Called the Processor")
+        try:
```
Don't catch workflow critical exceptions

ngr commented on lambdas/sys_user_management/samconfig.toml:
```diff
@@ -0,0 +1,10 @@
+version = 0.1
+
+[default.deploy.parameters]
+stack_name = "sosw-ms-user-management"
```
Rename stack

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,63 @@
+        # Generating password to users
+        for name in names:
+            password = self.generate_random_password()
+            try:
```
Don't catch. This is critical and should block

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,63 @@
+        return users
+
+
+    def generate_random_password(self, length=12):
```
8

ngr commented on lambdas/sys_user_management/template.yaml:
```diff
@@ -0,0 +1,53 @@
+      Handler: app.lambda_handler
+      MemorySize: 256
+      PackageType: Zip
+      Runtime: python3.11
```
3.12

too many empty strings

kreanimator committed:
Fix #3

kreanimator committed:
Fix #3

kreanimator committed:
Fix #3

kreanimator committed:
Added sys_user_management_cleanup and all related functions:
-delete_users_and_groups()
-delete_customer_managed_policies()
-delete_all_lambda_functions(region)
-delete_iam_roles()

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+class Processor(SoswProcessor):
+    DEFAULT_CONFIG = {
+        'init_clients':                ['iam', 'lambda'],
+        'cleanup_exclude_list_lambda': [],
```
- sys_user_management

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+    DEFAULT_CONFIG = {
+        'init_clients':                ['iam', 'lambda'],
+        'cleanup_exclude_list_lambda': [],
+        'cleanup_exclude_list_role':   ['AWSServiceRoleForComputeOptimizer ',
```
Trim Spaces.

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+
+        match trigger:
+            case ['create']:
+                self.sys_user_management(event)
```
rename function

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+        trigger = event.get('action', [])
+
+        match trigger:
+            case ['create']:
```
no need for list. Use string.

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+        match trigger:
+            case ['create']:
+                self.sys_user_management(event)
+                logger.info('Creating users')
```
action...ing = before method
action...ed = after

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+            function_name = function['FunctionName']
+            if function_name not in self.config['cleanup_exclude_list_lambda']:
+                self.lambda_client.delete_function(FunctionName=function_name)
+                logger.info('deleted lambda function', function_name)
```
logger.info('Deleted lambda function: %s', function_name)

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+                logger.info('deleted lambda function', function_name)
+
+                self.stats['lambdas_deleted'] += 1
+                self.stats['lambdas_deleted'].append(function_name)
```
You can not append sting to int

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+
+        for user in users:
+            username = user['UserName']
+            user_groups = self.iam_client.list_groups(UserName=users['UserName'])
```
Delete user groups at the end of the function and outside of the context of iterating users.

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -0,0 +1,151 @@
+    def delete_iam_roles(self):
+
+        response = self.iam_client.list_roles()
+        roles_to_die = response['Roles']
```
`roles = [x for x in response['Roles'] if x not in self.config['cleanup_exclude_list_role']]`

ngr commented on lambdas/sys_user_management/template.yaml:
```diff
@@ -0,0 +1,57 @@
+      MemorySize: 256
+      PackageType: Zip
+      Runtime: python3.11
+      Timeout: 10
```
Set timeout 300

kreanimator committed:
Fixed loggers, added uppercase to password generation, refactored code

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -32,22 +33,22 @@ def __call__(self, event):
+        trigger = event.get('action', '')
 
         match trigger:
             case ['create']:
```
your cases should not be lists, but strings.

kreanimator replied:
Ooops, sorry, forgot to clear brackets

ngr commented on lambdas/sys_user_management/src/app.py:
```diff
@@ -119,31 +117,26 @@ def delete_customer_managed_policies(self):
+            self.iam_client.delete_role(RoleName=role_name)
+            logger.info('Role successfully deleted: %s', role_name)
+            self.stats['roles_deleted'] += 1
+        else:
```
```
for i in _iterable_:
    pass
else:
    print("fires in case there was no 'break'")
```
Your message in the `else` statement will always fire since you do not break (and don't want to break) and can make some confusion. Thus it doesn't make much sense here. You better just remove it.

kreanimator committed:
Fixed loggers, added uppercase to password generation, refactored code
